tauri-app/src/lib/stores/settings.ts

```diff
@@ -4,59 +4,97 @@ import  find from 'lodash/find';
+  && v[1].network.label === $activeNetwork?.[1].label
+  && v[1].network.network_id === $activeNetwork?.[1].network_id
+  && v[1].network.currency === $activeNetwork?.[1].currency
+));
+export const activeOrderbookIndex = cachedWritableInt("settings.activeOrderbookIndex", 0);
+export const activeOrderbook =  derived([orderbooks, activeOrderbookIndex], ([$orderbooks, $activeOrderbookIndex]) => $orderbooks?.[$activeOrderbookIndex]);
+export const subgraphUrl = derived(activeOrderbook, ($activeOrderbookSettings) => $activeOrderbookSettings?.[1].subgraph);
+export const orderbookAddress = derived(activeOrderbook, ($activeOrderbookSettings) => $activeOrderbookSettings?.[1].address);
 
-export const hasRequiredSettings = derived([activeChainSettings, activeOrderbookSettings], ([$activeChainSettings, $activeOrderbookSettings]) => $activeChainSettings !== undefined && $activeOrderbookSettings !== undefined);
+export const hasRequiredSettings = derived([activeNetwork, activeOrderbook], ([$activeChainSettings, $activeOrderbookSettings]) => $activeChainSettings !== undefined && $activeOrderbookSettings !== undefined);
 
-// When settings data updated, reset active chain
-settings.subscribe((val) => {
-  if(val && val.chains.length < get(activeChainSettingsIndex)) {
-    activeChainSettingsIndex.set(0);
-  }
-});
+// deployments
+export const deployments = derived([settings, activeNetwork, activeOrderbook], ([$settingsData, $activeNetwork, $activeOrderbook]) => Object.entries($settingsData.deployments).filter(v => {
+  return v[1].order.network.rpc === $activeNetwork?.[1].rpc
+    && v[1].order.network.chain_id === $activeNetwork?.[1].chain_id
+    && v[1].order.network.label === $activeNetwork?.[1].label
+    && v[1].order.network.network_id === $activeNetwork?.[1].network_id
+    && v[1].order.network.currency === $activeNetwork?.[1].currency
+    && (
+      v[1].order.orderbook !== undefined
+        ? (
+          v[1].order.orderbook.address === $activeOrderbook?.[1].address
+          && v[1].order.orderbook.subgraph === $activeOrderbook?.[1].subgraph
+          && v[1].order.orderbook.label === $activeOrderbook?.[1].label
+        )
+        : true
+    )
+})
+);
+export const activeDeploymentIndex = cachedWritableInt("settings.activeDeploymentIndex", 0);
+export const activeDeployment = derived([deployments, activeDeploymentIndex], ([$deployments, $activeDeploymentIndex]) => $deployments?.[$activeDeploymentIndex]);
 
-// When active chain updated, reset active orderbook
-activeChainSettings.subscribe(async ()  => {
-  activeOrderbookSettingsIndex.set(0);
-});
+// // When networks data updated, reset active chain
+// networks.subscribe((val) => {
+//   if(val && val.length < get(activeNetworkIndex)) {
+//     activeNetworkIndex.set(0);
+//   }
+// });
+
+// // When active network updated, reset active orderbook and deployment
+// activeNetwork.subscribe(async ()  => {
+//   activeOrderbookIndex.set(0);
+//   activeDeploymentIndex.set(0);
+// });
```

Comment:
should this be commented out?

Reply:
well I would leave this part to you since you knwo better, I was just trying to get it work